core/arch/arm/kernel/kern.ld.S

```diff
@@ -379,7 +374,11 @@ SECTIONS
 
 }
 
-#if defined(CFG_TEE_COHERENT_START)
-ASSERT((__coherent_start & (4096 - 1)) == 0, "Coherent RAM start is not 4Kbyte aligned")
-ASSERT((__coherent_end & (4096 - 1)) == 0, "Coherent RAM end is not 4Kbyte aligned")
+#if CFG_TEE_COHERENT_SIZE
+ASSERT(!(__coherent_start & (4096 - 1)),
+					"Coherent memory start alignment");
+ASSERT(!(CFG_TEE_COHERENT_SIZE & (4096 - 1)),
```

Comment:
This and the following assert are useless, they are just checking that ` . = ALIGN(4096)` works.